### ISSUE TITLE
[Bug] Switching languages in-app doesn't override system language settings

### ISSUE THREAD

**Issue by gueroJeff** [mozilla-mobile/fenix#9290]
## Steps to reproduce

1. Open preferences.
2. Open language setting.
3. Select locale that is not part of the system languages (e.g., cak - Kaqchikel).

### Expected behavior
App UI should switch to the Kaqchikel localization.
### Actual behavior
Screen flashes, UI remains in preferred system language.
### Device information
Android 5.1.1
* Android device: ?
* Fenix version: build 20761805

**Comment by cadeyrn** [comment]
Do you use a dark Android and the light Fenix theme or a light Android and the dark Fenix theme? If yes, it's a duplicate of #7559 - the language switcher does not work if the Android and the Fenix theme do not match. I hope it will be fixed soon…

**Comment by gueroJeff** [comment]
I'm reporting this on behalf of a localizer. The screenshot he sent me appeared to be the light Fenix theme, but I can double check.

**Comment by eliserichards** [comment]
https://issuetracker.google.com/issues/136539680 - should be fixed in the latest version of `AppCompat`.

**Comment by ekager** [comment]
Reopening for QA

**Comment by abodea** [comment]
I tried to reproduce this issue but everything looks good now for more details please check the following:
The language now is correctly changed but only after restarting the app, please see the [reggression](https://github.com/mozilla-mobile/fenix/issues/9413).
Should I confirm once again and close this issue after #9413 will be verified?
I will remove the qa:needed until further notice.
@eliserichards @ekager

**Comment by liuche** [comment]
Since we have the other issue to track #9413, I think we can close this one!
